install/Knative-with-Minishift.md

```diff
@@ -0,0 +1,256 @@
+local machine using [minishift](https://docs.okd.io/latest/minishift/getting-started/index.html)
+to test-drive Knative.
+
+## Minishift setup
+
+- [Prepare virtualization environment](https://docs.okd.io/latest/minishift/getting-started/setting-up-virtualization-environment.html) to be used with Minishift
+- Download the archive for your operating system from the [Minishift v1.25.0](https://github.com/minishift/minishift/releases/tag/v1.25.0) Release page and extract its contents
+- Copy the `minishift` binary from extracted directory to your preferred location like /usr/local/bin
+- Ensure `minishift` is in your PATH by running the command:
+
+```shell
+# returns minishift v1.25.0+90fb23e
+minishift version
+```
+
+## Configure and start minishift
+
+The following details the bare minimum configuration required to setup minishift for running Knative:
+
+```shell
+
+# make sure you have  a profile is set correctly, e.g. knative
+minishift profile set knative
+
+# Pinning to the right needed OpenShift version, anything >= v3.11.0
+minishift config set openshift-version v3.11.0
+
+# minimum memory required for the minishift VM
+minishift config set memory 8GB
+
+# the minimum required vCpus for the minishift vm
+minishift config set cpus 4
+
+# extra disk size for the vm, default is 20g
+minishift config set disk-size 50g
+
+# caching the images that will be downloaded during app deployments
+minishift config set image-caching true
+
+# Add new user called admin with password admin having role cluster-admin
+minishift addons enable admin-user
+
+# Allow the containers to be run with uid 0
+minishift addons enable anyuid
+
+# start minishift
+minishift start
+```
+
+- The above configuration ensures that Knative gets created in its own [minishift profile](https://docs.okd.io/latest/minishift/using/profiles.html) called `knative` with 8GB of RAM, 4 vCpus and 50GB of hard disk. The image-caching helps in re-starting up the cluster faster every time.  
+- The [addon](https://docs.okd.io/latest/minishift/using/addons.html) **admin-user** creates a user called  `admin` with password  `admin` having the role of cluster-admin.  The user gets created only after the addon is applied, that is usually after successful start of Minishift
+- The  [addon](https://docs.okd.io/latest/minishift/using/addons.html) **anyuid** allows the `default` service account to run the application with uid `0`
+
+- The command `minishift profile set knative` is required every time you start and stop minishift to make sure that you are on right `knative` minishift profile that was configured above.
+
+## Configuring `oc` (openshift cli)
+
+Running the following command make sure that you have right version of `oc` and have configured the DOCKER daemon to be connected to minishift docker.
+
+```shell
+# configures the host talk to DOCKER daemon of minishift
+minishift docker-env
+# configures the right openshift cli
```

Comment:
s/configures the right openshift cli/Adds the right version of openshift cli binary to $PATH/